staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -88,6 +90,22 @@ func (agg aggregate) Error() string {
 	return "[" + result + "]"
 }
 
+type errMatcher interface {
+	Is(error) bool
+}
+
+func (agg aggregate) Is(target error) bool {
```

Comment:
The std lib code uses reflection to confirm that the `==` comparison is safe to perform and attempts to `Unwrap` the `error` as well:

```go
// Is reports whether any error in err's chain matches target.
//
// The chain consists of err itself followed by the sequence of errors obtained by
// repeatedly calling Unwrap.
//
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
func Is(err, target error) bool {
	if target == nil {
		return err == target
	}

	isComparable := reflectlite.TypeOf(target).Comparable()
	for {
		if isComparable && err == target {
			return true
		}
		if x, ok := err.(interface{ Is(error) bool }); ok && x.Is(target) {
			return true
		}
		// TODO: consider supporing target.Is(err). This would allow
		// user-definable predicates, but also may allow for coping with sloppy
		// APIs, thereby making it easier to get away with them.
		if err = Unwrap(err); err == nil {
			return false
		}
	}
}
```

Reply:
I think this is implicitly fixed now by following your other suggestion of re-using `errors.Is` rather than re-implementing it